graphrag/index/flows/create_base_text_units.py

```diff
@@ -32,15 +34,20 @@ def create_base_text_units(
 
     callbacks.progress(Progress(percent=0))
 
+    agg_dict = {"text_with_ids": list}
+    if "metadata" in documents:
+        agg_dict["metadata"] = "first"  # type: ignore
+
     aggregated = (
         (
             sort.groupby(group_by_columns, sort=False)
             if len(group_by_columns) > 0
             else sort.groupby(lambda _x: True)
         )
-        .agg(texts=("text_with_ids", list))
+        .agg(agg_dict)
         .reset_index()
     )
+    aggregated.rename(columns={"text_with_ids": "texts"}, inplace=True)
 
     aggregated["chunks"] = chunk_text(
```

Comment:
Do this as two steps so we don't need to edit the chunk_text method, something like:

`aggregated["chunks"] = chunk_text(...)`

```
if prepend_metadata:
  aggregated["chunks"] = aggregated["chunks"].apply(lambda x: meta + x)
```